### ISSUE TITLE
LayerPricer and InsurancePricer never cross-validated (ASOP 36 model validation)

### ISSUE THREAD

**Issue by AlexFiliakov** [AlexFiliakov/Ergodic-Insurance-Limits#1455]
## Problem Statement

The `LayerPricer` class (insurance_pricing.py, lines 187-348) and `InsurancePricer.calculate_pure_premium()` (lines 430-568) operate independently with no cross-validation between analytical and simulation-based pricing results.

`LayerPricer` computes expected layer losses analytically using LEV formulas:
```python
layer_loss = LEV(attachment + limit) - LEV(attachment)
```

`InsurancePricer` computes pure premium via Monte Carlo simulation of the loss generator.

These two methods should produce consistent results for the same layer and distribution, but:

1. **They are never compared**: No validation checks that analytical and simulation-based pricing agree within statistical tolerance.

2. **They may use different distributions**: `LayerPricer` accepts any `LossDistribution`, while `InsurancePricer` uses a `ManufacturingLossGenerator` which is a composite of multiple distributions. The composite aggregate cannot be compared against a single analytical distribution.

3. **No reconciliation guidance**: A user who prices a layer with both methods will get different numbers with no indication of which is more reliable or why they differ.

4. **Analytical pricing ignores aggregate effects**: `LayerPricer.expected_layer_loss()` returns `frequency * per_occurrence_layer_loss`, which is the expected annual aggregate for the layer assuming frequency-severity independence. But aggregate effects (reinstatements, annual aggregate limits, swing-rated features) are not captured in the analytical pricing.

## v1.0 Impact

**High** -- Having two pricing methods that may produce inconsistent results undermines user confidence. Cross-validation is a basic actuarial quality control step (ASOP 36, Section 3.2.4: validation of models).

## Affected Code

- `ergodic_insurance/insurance_pricing.py`: `LayerPricer` class, lines 187-348
- `ergodic_insurance/insurance_pricing.py`: `InsurancePricer.calculate_pure_premium()`, lines 430-568

## Current Behavior

`LayerPricer` and `InsurancePricer` are completely independent classes. No cross-validation, no reconciliation, no guidance on when to use which.

## Expected Behavior

1. When both methods are available, automatically cross-validate and warn if results diverge beyond tolerance
2. Document when analytical vs. simulation pricing is appropriate
3. Provide a `reconcile()` method that compares both approaches for the same layer

## Recommended Approach

Add a `validate_against_analytical()` method to `InsurancePricer` that:
1. Creates a `LayerPricer` from the same severity distribution (when available)
2. Compares analytical and simulation pure premiums
3. Reports the ratio and flags divergence beyond a configurable tolerance (default 10%)
4. Warns if the simulation sample size is too small for reliable comparison

## Acceptance Criteria

- [ ] Cross-validation method compares analytical and simulation pricing
- [ ] Warning emitted when methods diverge beyond configurable tolerance
- [ ] Documentation explains when each method is appropriate (analytical for simple layers, simulation for aggregate features)
- [ ] Unit test confirms methods agree within tolerance for a simple lognormal layer

**Comment by AlexFiliakov** [comment]
**Validation review (2026-07-01):** The gap is real and unchanged — `LayerPricer` (insurance_pricing.py:187) and `InsurancePricer.calculate_pure_premium()` (insurance_pricing.py:430) remain fully independent, with no `reconcile()`/`validate_against_analytical()` anywhere in the module. However, this duplicates open issue #1274 ("LayerPricer analytical pricing and InsurancePricer simulation pricing never cross-validated"), which predates it and carries priority-medium. The medium priority is the honest label: this is a missing quality-control feature, not a defect making either pricing path incorrect, so it does not block v1.0 calculation accuracy. Note also the standards citation is off — model validation guidance is ASOP 56 (Modeling); ASOP 36 concerns P/C reserve opinions. Recommend the maintainer consolidate with #1274 and align priority at medium.
